File: app/src/androidTest/java/org/tasks/caldav/SynologyTests.java
package org.tasks.caldav;

import static junit.framework.Assert.assertTrue;
import static org.tasks.TestUtilities.vtodo;

import androidx.test.runner.AndroidJUnit4;
import java.util.TimeZone;
import org.junit.After;
import org.junit.Before;
import org.junit.Test;
import org.junit.runner.RunWith;

@RunWith(AndroidJUnit4.class)
public class SynologyTests {

  private TimeZone defaultTimeZone = TimeZone.getDefault();

  @Before
  public void before() {
    TimeZone.setDefault(TimeZone.getTimeZone("America/Chicago"));
  }

  @After
  public void after() {
    TimeZone.setDefault(defaultTimeZone);
  }

  @Test
  public void completedWithoutDueDate() {
    assertTrue(vtodo("synology/complete_no_due_date.txt").isCompleted());
  }

  @Test
  public void completedWithDueDate() {
    assertTrue(vtodo("synology/complete_with_date.txt").isCompleted());
  }

  @Test
  public void completedWithDateTime() {
    assertTrue(vtodo("synology/complete_with_date_time.txt").isCompleted());
  }
}

File: app/src/androidTest/java/org/tasks/caldav/SynologyTests.kt
package org.tasks.caldav

import androidx.test.ext.junit.runners.AndroidJUnit4
import org.junit.After
import org.junit.Assert.assertTrue
import org.junit.Before
import org.junit.Test
import org.junit.runner.RunWith
import org.tasks.TestUtilities.vtodo
import java.util.*

@RunWith(AndroidJUnit4::class)
class SynologyTests {
    private val defaultTimeZone = TimeZone.getDefault()

    @Before
    fun before() {
        TimeZone.setDefault(TimeZone.getTimeZone("America/Chicago"))
    }

    @After
    fun after() {
        TimeZone.setDefault(defaultTimeZone)
    }

    @Test
    fun completedWithoutDueDate() {
        assertTrue(vtodo("synology/complete_no_due_date.txt").isCompleted)
    }

    @Test
    fun completedWithDueDate() {
        assertTrue(vtodo("synology/complete_with_date.txt").isCompleted)
    }

    @Test
    fun completedWithDateTime() {
        assertTrue(vtodo("synology/complete_with_date_time.txt").isCompleted)
    }
}